docs/docs/wikitext103.md

```diff
@@ -0,0 +1,205 @@
+# WikiText-103
+
+![Banner](img/banner.png)
+[//]: # (TODO Change Me) 
+
+You can download WikiText-103 dataset [here](https://blog.einstein.ai/the-wikitext-long-term-dependency-language-modeling-dataset/), there is no official leaderboard.
+
+## TLDR;
+
+Here is a minimal working example that let you evaluate `transformer-xl` from `huggingface/transformers`.
+
+```python
+import torch
+from sotabencheval.langauge_modelling import WikiText103Evaluator
+
+model = torch.hub.load('huggingface/transformers', 'modelWithLMHead', 'transfo-xl-wt103').to("cuda")
+tokenizer = torch.hub.load('huggingface/transformers', 'tokenizer', 'transfo-xl-wt103').to("cuda")
+
+with WikiText103Evaluator.dataset.get_path('local/dir/wikitext-103').open() as f:
```

Comment:
please use `is_server()` so it's not just using local paths (see coco/imagenet docs). Also the use of a context manager here is a bit confusing, this needs to be documented (and if API can be simpler that's great).

Reply:
The `get_path` method has the following signature `(local_path:StrOrPath) -> Path`. The `is_server` is being used in the `get_path` method, and the path to the `wikitext-103-v1.zip`  is hardcoded in the method, along with the use of `extract_archive` that unpack the zip if needed.

The context manager and `.open` is the idiom from `pathlib.Path`. 
We have the following ways already implemented to get path to the dataset all of them are taking care of extracting the archive let me know which are better:

If you want to read the dataset before you instantiate the evaluator you can do this on 3 ways:
Using is_server and rewrting the logic of get_path:
````
if is_server():
      root = Path(".data/nlp/wikitext-103")
else:
      root = Path("/path/to/your/location/where/you/have/unzipped/wikitext103")
dataset_path = root / "wiki.test.tokens"
if not dataset_path.exists(): 
       # on server we need to unzip the wikitext first.
       # note the use of the root.parent as the .zip file has already wikitext-103 folder in it
       # without this root.parent you need to set the path to `.data/nlp/wikitext-103/wikitext-103/wiki.test.tokens `
       extract_archive(str(root / "wikitext-103-v1.zip"), to_path=root.parent)

with dataset_path.open() as f:
    testset_content = f.read()
```
Or we can simplify the code using the convenience method of `dataset.get_path` 
```
path_to_testset =  WikiText103Evaluator.dataset.get_path(local_root="/path/to/your/location/where/you/have/unzipped/wikitext103")
with path.open() as f:
    testset_content = f.read()
```

The dataset is actually an instance of WikiTextDataset enum so we could use that directly:
```
path_to_testset = WikiTextDataset.Wikitext103.get_path(local_root="/path/to/your/location/where/you/have/unzipped/wikitext103")
with path.open() as f:
    testset_content = f.read()
```

Things might be even simpler if we read the dataset after the WIkiText103Evaluator was created:
```
   evaluator= WikiText103Evaluator(local_root="/path/to/your/local/data", model_name ...)
   with  evaluator.dataset_path.open() as f:
         testset_content = f.read()
```

Given that we may want to evaluate on Validation set as well we may want to return paths and change the `dataset_path` property to `test_set_path` or `validation_set_path`

And change the `WikiTextDataset.WikiText103.get_path` to `get_test_set_path` and `get_validation_set_path`.

I'll ask on slack which api is better and the right example in to the docs.

Reply:
Yep I think something like this makes sense. Makes sense that you need something like this for language modelling where the user needs more control over the data - e.g. needs to tokenise the corpus with their own logic.

Just needs to be documented clearly - I think a step-by-step example is the best way to explain why we need this.